homeassistant/components/meteo_france/sensor.py

```diff
@@ -23,52 +26,49 @@
 STATE_ATTR_BULLETIN_TIME = "Bulletin date"
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Meteo-France sensor."""
-    if discovery_info is None:
-        return
-
-    city = discovery_info[CONF_CITY]
-    monitored_conditions = discovery_info[CONF_MONITORED_CONDITIONS]
-    client = hass.data[DATA_METEO_FRANCE][city]
-    weather_alert_client = hass.data[DATA_METEO_FRANCE]["weather_alert_client"]
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the Meteo-France sensor platform."""
+    city = entry.data[CONF_CITY]
+    client = hass.data[DOMAIN][city]
+    weather_alert_client = hass.data[DOMAIN]["weather_alert_client"]
 
     alert_watcher = None
-    if "weather_alert" in monitored_conditions:
-        datas = hass.data[DATA_METEO_FRANCE][city].get_data()
-        # Check if a department code is available for this city.
-        if "dept" in datas:
-            try:
-                # If yes create the watcher DepartmentWeatherAlert object.
-                alert_watcher = DepartmentWeatherAlert(
-                    datas["dept"], weather_alert_client
-                )
-            except ValueError as exp:
-                _LOGGER.error(
-                    "Unexpected error when creating the weather alert sensor for %s in department %s: %s",
-                    city,
-                    datas["dept"],
-                    exp,
-                )
-                alert_watcher = None
-            else:
-                _LOGGER.info(
-                    "Weather alert watcher added for %s in department %s",
-                    city,
-                    datas["dept"],
-                )
+    datas = client.get_data()
+    # Check if a department code is available for this city.
+    if "dept" in datas:
+        try:
+            # If yes create the watcher DepartmentWeatherAlert object.
+            alert_watcher = await hass.async_add_executor_job(
+                DepartmentWeatherAlert, datas["dept"], weather_alert_client
+            )
+        except ValueError as exp:
+            _LOGGER.error(
+                "Unexpected error when creating the weather alert sensor for %s in department %s: %s",
+                city,
+                datas["dept"],
+                exp,
+            )
+            alert_watcher = None
```

Comment:
Are we going to pass `alert_watcher` as `None`? Will this recover later, or does this mean the entity will not use this feature?

Reply:
The entity will not use this feature (only used in `weather_alert` sensor perhaps).

Some "department" has no weather alert from Météo-France.